### PR TITLE
Fixed disappearing notice Cloudflare

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description
Fix an issue with the error disappearing if we save the page without changing any value. 

For that we fixed a couple of errors and change the check for the error to the `pre_update` filter.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Locally

# Checklist:

Please delete the options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: inc/Addon/Cloudflare/Subscriber.php [hunk 1]
```diff
@@ -440,25 +440,25 @@ private function save_cloudflare_auto_settings( $auto_settings, $old_settings )
 	 * @param array $old_value An array of previous values for the settings.
 	 * @param array $value     An array of submitted values for the settings.
 	 *
-	 * @return void
+	 * @return array
```
Comment (remyperona) [inline inc/Addon/Cloudflare/Subscriber.php]: `update_option_` is an action, callback method doesn't need any returned value, why this change?

File: inc/Addon/Cloudflare/Subscriber.php [hunk 2]
```diff
@@ -485,25 +487,25 @@ public function update_dev_mode( $old_value, $value ) {
...
```
Comment (remyperona) [inline inc/Addon/Cloudflare/Subscriber.php]: same question

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@CrochetFeve0251 Thanks for the update. Saving the settings while the error is displayed won't hide the error. However,  please find exploratory test notes below:
- Refreshing the page while the error is displayed, will hide the error.  => Shouldn't the error still be displayed here?
- If we have 3.14, where the error was hidden due to the bug there, then we update to the PR, shouldn't the error be displayed? (currently, it is not)

**Comment by CrochetFeve0251** [comment]
> @CrochetFeve0251 Thanks for the update. Saving the settings while the error is displayed won't hide the error. However,  please find exploratory test notes below:
> - Refreshing the page while the error is displayed, will hide the error.  => Shouldn't the error still be displayed here?
> - If we have 3.14, where the error was hidden due to the bug there, then we update to the PR, shouldn't the error be displayed? (currently, it is not)

@Mai-Saad the error is displayed when we save so it is normal if we don't save anything.
Are we supposed to display an error even nothing is currently saved?

**Comment by Mai-Saad** [comment]
@CrochetFeve0251 As per discussion with @piotrbak nothing is needed here and fine as is.

**Review by Mai-Saad — APPROVED** [review]
[testrail-report-494.pdf](https://github.com/wp-media/wp-rocket/files/11788479/testrail-report-494.pdf)
Working as expected with the following note (agreed to remain as is)
The 1st time we save settings while having wrong CF addon data + CF plugin is on and authenticated => validation error is displayed and any other save settings after that won't show the error

**Pull request merged**